snap/frontend/utils/snap.js

```diff
@@ -4,6 +4,24 @@ const symbolSnapFactory = {
 	create(provider) {
 		return {
 			provider: provider,
+			/**
+			 * A helper function to invoke a method in the snap.
+			 * @param {string} method - The method to invoke.
+			 * @param {object} params - The params to pass with the method.
+			 * @returns {object} The response from the snap.
+			 */
+			async invokeSnapMethod(method, params = {}) {
+				return await this.provider.request({
+					method: 'wallet_invokeSnap',
+					params: {
+						snapId: defaultSnapOrigin,
+						request: {
+							method,
+							...(Object.keys(params).length && { params })
```

Comment:
Do you need the ``keys().length`` check? If the params are empty, then nothing gets copied.

Reply:
yes, if `params` is empty, I will skip the `params`.